src/thunderbots/software/radio_communication/visitor/radio_packet_serializer_primitive_visitor.h

```diff
@@ -0,0 +1,77 @@
+#pragma once
+
+#include "ai/primitive/visitor/primitive_visitor.h"
+
+/**
+ * This class implements a Visitor that serializes the Primitive classes into packets
+ * that can be send over the radio to the physical robots
+ */
+class RadioPacketSerializerPrimitiveVisitor : public PrimitiveVisitor
+{
+   public:
+    /**
+     * Creates a new RadioSerializerPrimitiveVisitor
+     */
+    RadioPacketSerializerPrimitiveVisitor() = default;
+
+    /**
+     * Serializes the given CatchPrimitive into a radio packet
+     *
+     * @param catch_primitive The CatchPrimitive to serialize
+     */
+    void visit(const CatchPrimitive &catch_primtiive) override;
+
+    /**
+     * Serializes the given ChipPrimitive into a radio packet
+     *
+     * @param chip_primitive The ChipPrimitive to simulate
+     */
+    void visit(const ChipPrimitive &chip_primtiive) override;
+
+    /**
+     * Serializes the given DirectVelocityPrimitive into a radio packet
+     *
+     * @param direct_velocity_primitive The DirectVelocityPrimitive to simulate
+     */
+    void visit(const DirectVelocityPrimitive &direct_velocity_primtiive) override;
+
+    /**
+     * Serializes the given KickPrimitive into a radio packet
+     *
+     * @param kick_primitive The KickPrimitive to simulate
+     */
+    void visit(const KickPrimitive &kick_primtiive) override;
+
+    /**
+     * Serializes the given MovePrimitive into a radio packet
+     *
+     * @param move_primitive The MovePrimitive to simulate
+     */
+    void visit(const MovePrimitive &move_primitive) override;
+
+    /**
+     * Serializes the given PivotPrimitive into a radio packet
+     *
+     * @param pivot_primitive The PivotPrimitive to simulate
+     */
+    void visit(const PivotPrimitive &pivot_primtiive) override;
+
+    /**
+     * Returns the most recent serialized packet created by this
+     * RadioPacketSerializerPrimitiveVisitor. This is the radio packet created by
+     * one of the 'visit' functions
+     *
+     * Calling this function before this Visitor has been called at least once has
+     * undefined behavior
```

Comment:
:nauseated_face: could we return an optional instead?

Reply:
Yep. I'll change this for the `GrSimCommandPrimitiveVisitor` too